### DISCUSSION TITLE
How to configure .gitignore when using auto-generated content

### DISCUSSION THREAD

**Discussion by grant-allan-ctct** [sphinx-doc/sphinx#12221] [Q&A]
Hello, I am wondering if there is some official guidance on what to put in .gitignore when using modern-day Sphinx with a Python project?   

I'm just getting started with Sphinx and it looks great already, by the way!!  

I have looked at the documentation [here](https://www.sphinx-doc.org/en/master/tutorial/deploying.html#publishing-your-documentation-sources) and it mentions to push "everything under `docs/source`" to the repo.  

But I see that I currently have a `generated` folder in my `docs/source` folder and I am curious about whether that one (which I presume got built for me by one of the Sphinx extensions that I have tried out - either `sphinx.ext.autodoc` or `sphinx.ext.autosummary`) maybe should _not_ be added to the repo?

I also see a `_static` and `_templates` folder in my `docs/source` folder, both which are empty folders.  Is it best to push those to the repo, or is the leading underscore a hint that it contains transient content and is not something to be version-controlled?  In looking at my `conf.py` file, I can see them both mentioned in there, but I confess to not knowing what they are:

```
templates_path = ['_templates']
html_static_path = ['_static']

```

Many thanks for your advice!

**Comment by jdillard** [comment]
In general, it isn't best practice with static site generators to commit built and/or auto-generated content to the repo, they get built and deployed as part of a continuous deployment pipeline.
